lib/fauna/client.rb

```diff
@@ -237,6 +237,8 @@ def perform_request(action, path, query, data)
         req.body = FaunaJson.to_json(data) unless data.nil?
         req.url(path || '')
       end
+    rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
```

Comment:
You should also rescue any [Faraday::Error](https://github.com/lostisland/faraday/blob/80c0e66102168939df04682573bc267e089a43ae/lib/faraday/error.rb#L2) and throw a `UnexpectedError` if status is not 503.

Reference: https://github.com/faunadb/faunadb-jvm/blob/master/faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java#L257